### PR TITLE
Building day filter function

### LINKED ISSUE #279: filter bus timetable dataframe for a certain date
Create a function that filters the linked dataframe created in #278 for a user specified date.

### PR DESCRIPTION
**Pull request by jwestw**
- Created a filter by day function
- `day` is a parameter in the function

Closes #279

### CODE REVIEW

**Comment by nkshaw23** [comment]
Added edits to `best_before` to check for modified date rather than creation date. 

Changed logic on the `filter_bus_timetable_by_day` function so records are now returned. Ran with `ord=14` (right in the middle of the start and end dates) and it returned 17 million records. Also tried with `ord = 1,7,21,28` to test a range a of value and these returned very few records.

So, it seems like chossing a day that is central between `start_date` and `end_date` is key. 

We could force this in the code by setting `ord` equal to the `value_count` in `days_counted` for the day specified.

**Comment by jwestw** [comment]
@Antonio-John @paigeh1 Please can you please have a look over this code? It's not a critical merge because I am not merging back into main, but still work checking. 

Nathan is away for a couple of weeks so I'll need a bit of assistance. Thanks

File: SDG_bus_timetable.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 # Parameters
 bus_timetable_zip_link = config["ENG_bus_timetable_data"]
 bus_dataset_name = 'itm_all_gtfs'
-output_directory = os.path.join(CWD, 'data', 'england_bus_timetable')
+DATA_DIR = di.DATA_DIR
```
Comment (Antonio-John) [inline SDG_bus_timetable.py:25]: Read in through config to be consistent with other constants.

**Review by Antonio-John — CHANGES_REQUESTED** [review]
-In line 35 & 84, 93, 104 where ```_persistent_exists``` or ```_csv_to_df``` is used, in futue any_to_pd could be used in line with best practice #313 

-The bus timetable is downloaded each time automatically. Instead of this it could be downloaded manually once outside of the script to prevent downloading issues in the future.

-In line 114 the ```valid_hours``` could be extended to only include hours between 6am and 8pm as per the calculation methodology.  

- ```bus_timetable_df``` uses a double merge. This could be split into 2 separate to be more readable. 

-Misspelling of Downloading in 208 in data_ingest.py

-The final dataframe ```serviced_bus_stops``` has days of the weeks column e.g monday with values of 1 or 0. Where has this come from?

**Comment by james-westwood** [comment]
The days columns are in the original calendars.txt / csv

image

This *might* mean we have an opportunity to filter by day at a much earlier stage (and much more simply). What we have implemented works for now though. I suggest optimising later if needs be as this will not be part of the main pipeline

**Comment by james-westwood** [comment]
Hey @nkshaw23 I am hoping you can pick the last remaining changes needed to get this PR done while I am away. I don't think there's much left to do to get it over the line. 
Antonio has reviewed this (with my help) and will be somewhat familiar with the code now, so he might be the best person (again with help from you) to carry out the final review. 

I edited Antonio's [review](https://github.com/ONSdigital/SDG_11.2.1/pull/312#pullrequestreview-1118095768) to easily reference the issues.

1) Don't do this. I spent a while messing around with `any_to_pd` and while I believe that this is the way to go. 
2) I decided to hide the download functions behind a flag, `auto_download_bus` which is in the config. We need to create a ticket to update the instructions to manually download the zip and update the name in the config file so it can be found. I would err towards default being that auto the download is `False`. Please create  a ticket for the documentation update. 
3) The filtering is now occuring on the actual valid hours (6am to 8pm). I don't love the way I have implemented the function. Check the most recent commit and the config

As a bonus, the dataset is now loading from a feather which speeds the read in up by about 10x!

**Comment by nkshaw23** [comment]
@Antonio-John Been through this and made some changes. Tested my end. Would you be able to give another review?

1) Removed code James added in to get config and filepaths working on his vm. This was giving mixed slashes in filepaths, which were consequently not being recognised. Used approach in SDG_main with hard coding directories into `os.path.join`.

2) Removed some unused imports, parameters and functions.

3) Removed `filter_by_year` call because this removes alot of records (almost 2/3rds). This is because any service with a timetable that runs into 2023 will be removed. The `filter_by_day` function will include these services.

4) Updated `ord` value in `filter_by_day` to be calculated as roughly the halfway point rather than `14` by default. If data is re-downloaded in the future, `14` might not be suitable, whereas the halfway point always will be.

**Pull request merged**